src/UnitySampleProject/Assets/Samples/web3.unity SDK Lootboxes/3.0.6/Lootbox Samples/Scripts/RewardsManager.cs

```diff
@@ -0,0 +1,79 @@
+using System.Globalization;
+using ChainSafe.Gaming.Lootboxes.Chainlink;
+using Nethereum.Util;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class RewardsManager : MonoBehaviour
+{
+    [SerializeField] private GameObject rewardsObjectPrefab;
+    [SerializeField] private Transform rewardsContainer;
+    [SerializeField] private Button closeRewardsButton;
+
+    private void OnEnable()
+    {
+        EventManager.ToggleRewardItems += SpawnRewards;
+    }
+
+    private void OnDisable()
+    {
+        EventManager.ToggleRewardItems -= SpawnRewards;
+    }
+
+    private void Awake()
+    {
+        closeRewardsButton.onClick.AddListener(CloseRewardsMenu);
+    }
+
+    private void CloseRewardsMenu()
+    {
+        Debug.Log("Close rewards menu");
+        ClearPreviousRewards();
+        gameObject.SetActive(false);
+    }
+
+    private void ClearPreviousRewards()
+    {
+        foreach (Transform child in rewardsContainer)
+        {
+            Destroy(child.gameObject);
+        }
+    }
+
+    private void SpawnRewards(LootboxRewards rewards)
+    {
+        // Clear previous items
+        ClearPreviousRewards();
+
+        // Loop through and spawn rewards based on type
+        foreach (var reward in rewards.Erc20Rewards)
+        {
+            var balance = UnitConversion.Convert.FromWei(reward.AmountRaw);
+            SpawnRewardItem("ERC20", "Tokens", "ERC20", balance.ToString(CultureInfo.InvariantCulture));
+        }
+        foreach (var reward in rewards.Erc721Rewards)
+        {
+            SpawnRewardItem("ERC721", $"#{reward.TokenId.ToString()}", reward.TokenName, "1");
+        }
+        foreach (var reward in rewards.Erc1155Rewards)
+        {
+            SpawnRewardItem("ERC1155", $"#{reward.TokenId.ToString()}", reward.TokenName, reward.Amount.ToString());
+        }
+    }
+
+    // Helper method to create and populate a reward item UI element
+    private void SpawnRewardItem(string type, string tokenId, string tokenName, string amount)
+    {
```

Comment:
This is not done per the best practices we've talked about several times now already 😢

Reply:
fixed